src/java.desktop/windows/classes/sun/print/PrintServiceLookupProvider.java

```diff
@@ -146,16 +147,20 @@ private synchronized void refreshServices() {
             }
         }
 
+        invalidateServices();
+        printServices = newServices;
+    }
+
+    private void invalidateServices() {
         // Look for deleted services and invalidate these
         if (printServices != null) {
             for (int j=0; j < printServices.length; j++) {
                 if ((printServices[j] instanceof Win32PrintService) &&
-                    (!printServices[j].equals(defaultPrintService))) {
+                            (!printServices[j].equals(defaultPrintService))) {
```

Comment:
Is it not right?
I admit I don't understand what you mean by _map_ here.

When the code is written like it was:
```
                if ((printServices[j] instanceof Win32PrintService) &&
                    (!printServices[j].equals(defaultPrintService))) {
                    ((Win32PrintService)printServices[j]).invalidateService();
                }
```
it's hard to scan: it's not clear what is part of the condition and what is the statement inside the if block.

I'd prefer to write it like this:
```
                if ((printServices[j] instanceof Win32PrintService)
                    && (!printServices[j].equals(defaultPrintService))) {

                    ((Win32PrintService)printServices[j]).invalidateService();
                }
```
That is moving the operator to the continuation line which makes it obvious it is a continuation line of the condition and adding a blank line before the statement in the code. It's still not perfect, however; and it changes the author in `blame` output.

I indented the continuation line by additional 8 spaces, which is also a common practice in Java, to visually separate the condition and the statement. In fact, it's IDE that updated the formatting, I decided to keep it because it makes the code clearer.

I can revert the change to this line if you like.
Or I can just add a blank line between the condition and the statement.

What is your preference?

Reply:
By mapping i mean same indentation for all conditions in if statement without adding additional indentation for each continuation line like

if ((condition1) &&
     (condition2)) {
}

or 

if ((condition1)
     &&(condition2)) {
}

I have not come across code in java.desktop where we add indentation at each continuation line of 'if' condition.
I understand difficulty to scan without indentation but then in cases where we have multiple lines on continuation line in if statement we will easily hit 80 characters limit.

If we want to differentiate between if conditions and actual statement execution to improve readability, we can move the statement block to new line like 

if ((condition1) &&
     (condition2))
{
}

Reply:
I would prefer if you revert this line or if we want to put emphasis on readability moving '{' to new line also seems fine.

Reply:
> By mapping i mean same indentation for all conditions in if statement…
Okay, it's the first time I've come across to such a use of `map`.

> I have not come across code in java.desktop where we add indentation at each continuation line of 'if' condition.
Take a look at [DefaultTableCellRenderer.getTableCellRendererComponent](https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/javax/swing/table/DefaultTableCellRenderer.java#L201).

> I understand difficulty to scan without indentation but then in cases where we have multiple lines on continuation line in if statement we will easily hit 80 characters limit.
What is more important code readability or the strict limit of 80 columns?

All in all, all these styles are used throughout `java.desktop` module.
I chose to opt for readability in this particular case and the line fits into 80 column limit.

Do I revert the change to this line?
Any other suggestion? What is your preference?

Reply:
> I would prefer if you revert this line or if we want to put emphasis on readability moving '{' to new line also seems fine.

Done.

Reply:
I do not suggest that the change should be moved forth and back, but I think that the second conditions should always be shifted, and if this causes 80 chars overflow then some other line split/rename/etc should be done to prevent that.

Recent example of such style...:
            if ((getColorSpaceType (p) == ColorSpace.TYPE_RGB) &&
                (getData (p, icSigMediaWhitePointTag) != null) &&
                (getData (p, icSigRedColorantTag) != null) &&
                (getData (p, icSigGreenColorantTag) != null) &&
                (getData (p, icSigBlueColorantTag) != null) &&
                (getData (p, icSigRedTRCTag) != null) &&
                (getData (p, icSigGreenTRCTag) != null) &&
                (getData (p, icSigBlueTRCTag) != null)) {
                thisProfile = new ICC_ProfileRGB (p);

Is that really looks fine?

Reply:
This looks even worse than the case in this PR which has only two conditions.

If it'd been the new code, I wouldn't have agreed to revert indentation.

Let's wait for another opinion… 

To make it clear, you're for keeping the indentation as it was in the original PR to visually separate condition from the statement in the if block. Do I get it right, @mrserb?

Reply:
> To make it clear, you're for keeping the indentation as it was in the original PR to visually separate condition from the statement in the if block. Do I get it right, @mrserb?

I think it looks better.

Reply:
BTW another solution is to move { to the next line if the previous statement was split across few lines.

Reply:
I added a blank line between the condition and the body.
Does it look good? @mrserb, @jayathirthrao

Reply:
[Code Conventions for Java](https://www.oracle.com/java/technologies/javase/codeconventions-indentation.html#248) say, “Line wrapping for `if` statements should generally use the 8-space rule, since conventional (4 space) indentation makes seeing the body difficult.” (It's the second to last block on the page.)

Reply:
If we are adding a new line, then I think we should need to add at l129, l138 otherwise it will look odd doing it at one place only.

Reply:
I haven't touched that code at all.
Not that odd because it's isolated to the new function now.

What is your suggestion? Refactor all if statements in these two functions? Submit a separate bug for refactor all if statements the entire file? Revert back to no new line, leaving this particular if untouched as well?